desktop/libs/indexer/src/indexer/api3_tests.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python
+# Licensed to Cloudera, Inc. under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  Cloudera, Inc. licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import sys
+from nose.tools import assert_equal
+from django.utils.datastructures import MultiValueDict
+from django.core.files.uploadhandler import InMemoryUploadedFile
+
+from desktop.settings import BASE_DIR
+from indexer.api3 import upload_local_file
+
+if sys.version_info[0] > 2:
+  from urllib.parse import unquote as urllib_unquote
+  from unittest.mock import patch, Mock, MagicMock
+else:
+  from urllib import unquote as urllib_unquote
+  from mock import patch, Mock, MagicMock
+
+
+def test_xlsx_local_file_upload():
+
+  csv_file = '''test 1,test.2,test_3,test_4
```

Comment:
Since using triple quotes ('''), looks like it requires `\n` for every expected next line @agl29.